cmd/collectors/rest/rest.go

```diff
@@ -464,25 +478,20 @@ func (r *Rest) processEndPoints(data *matrix.Matrix) error {
 				}
 
 				for _, metric := range endpoint.prop.metrics {
-					f := instanceData.Get(metric)
+					metr, ok := data.GetMetrics()[metric.name]
+					if !ok {
+						if metr, err = data.NewMetricFloat64(metric.name); err != nil {
+							r.Logger.Error().Err(err).
+								Str("name", metric.name).
+								Msg("NewMetricFloat64")
+						}
+					}
+					f := instanceData.Get(metric.name)
```

Comment:
when err != nil, is metr valid?

Reply:
metr is not added to matrix if there is an error. In current case, we check for duplicates.